webapp/views.py

```diff
@@ -87,6 +90,22 @@ def releasenotes_redirect():
         return flask.redirect(f"https://wiki.ubuntu.com/Releases")
 
 
+def search_snaps():
+    """
+    A JSON endpoint to search the snap store API
+    """
+
+    query = flask.request.args.get("q", "")
+    arch = flask.request.args.get("arch", "amd64")
+    size = flask.request.args.get("size", "100")
+    page = flask.request.args.get("size", "1")
+
+    if not query:
+        flask.abort(400)
```

Comment:
since it's a json endpoint maybe just:

`return flask.jsonify({"error": "No query"}), 400`